### DISCUSSION TITLE
Failed to initialize kinesis video with an exception: Unable to create Iot Credential provider

### DISCUSSION THREAD

**Discussion by ajaypsiborg** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1065] [Q&A]
INFO ] [14-09-2023 13:12:54:928.659 GMT] Using region: ap-south-1
[INFO ] [14-09-2023 13:12:54:928.910 GMT] Using IoT credentials for Kinesis Video Streams
[DEBUG] [14-09-2023 13:12:54:932.110 GMT] Creating IoT auth callbacks.
2023-09-14 13:12:55.081 ERROR   blockingCurlCall(): Curl call response failed with http status 403
[ERROR] [14-09-2023 13:12:55:081.844 GMT] Unable to create Iot Credential provider. Error status: 0x16000001
[ERROR] [14-09-2023 13:12:55:082.132 GMT] Failed to initialize kinesis video with an exception: Unable to create Iot Credential provider. Error status: 0x16000001

**Comment by sirknightj** [comment]
Hi @ajaypsiborg, please help us by providing the following:

1. What GStreamer command are you using? (with credentials redacted, if any)
3. What does your policy look like?
4. Are you able to fetch the credentials manually by making the curl call, then using the temporary credentials obtained to make a describe call?  Step 4, points 2 and 3 https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/how-iot.html#how-iot-test-it,

**Comment by ajaypsiborg** [comment]
start uploading live video streaming with usb web cam but not with raspberry pi camera 

Getting error in kinesis
StartTimestamp must be before the current time for LIVE_REPLAY Playback Mode.
Status code: 400
Request ID: cb574964-c928-4e9a-8241-f8fbab32b672
Region: ap-south-1
![test](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/assets/145026073/1b190f5e-5f58-44db-a9b8-a477a878e200)

Time: September 15, 2023 6:13:36 PM
please check the attached screen shot successfully uploading live stream using raspberry cam but not showing in kinesis but working fine with usb camera so please suggest

**Comment by niyatim23** [comment]
Hi @ajaypsiborg, the error code you're seeing is `0x16000001: STATUS_CURL_PERFORM_FAILED`. I think the iot credentials are not configured on your device correctly on your raspberry pi. Please make sure the curl call @sirknightj mentioned works before you try to send media with KVS

**Comment by disa6302** [comment]
@ajaypsiborg , 

For the raspberry pi camera, we need logs :)
The screenshot you have attached seems to be the working case.
